packages/web3-net/src/rpc_method_wrappers.ts

```diff
@@ -1,31 +1,25 @@
+import { DataFormat, format } from 'web3-common';
 import { Web3Context } from 'web3-core';
-
-import { convertToValidType, ValidReturnTypes, ValidTypes } from 'web3-utils';
 import * as rpcMethods from './rpc_methods';
 import { Web3NetAPI } from './web3_net_api';
 
-export async function getId<ReturnType extends ValidTypes = ValidTypes.HexString>(
+export async function getId<ReturnFormat extends DataFormat>(
 	web3Context: Web3Context<Web3NetAPI>,
-	returnType?: ReturnType,
+	returnFormat: ReturnFormat,
 ) {
 	const response = await rpcMethods.getId(web3Context.requestManager);
 
-	return convertToValidType(
-		response,
-		returnType ?? web3Context.defaultReturnType,
-	) as ValidReturnTypes[ReturnType];
+	return format({ eth: 'uint' }, response as unknown as number, returnFormat);
```

Comment:
```suggestion
	return format({ eth: 'uint' }, response as unknown as string, returnFormat);
```

The version (id) is returned as a number string

Reply:
Same as above, the value is identified and casted by schema, type here is used to properly format the output type.